### DISCUSSION TITLE
Auto floating

### DISCUSSION THREAD

**Discussion by artemave** [i3/i3#5925] [Show and tell]
One can set certain apps to start in floating mode in i3 config, but that's a bit too manual. Wouldn't it be nice if i3 just remembered my choice when I turn on floating and then apply it next time I start the same app? Well, i3 can't do that, but [this thing](https://github.com/artemave/i3_auto_floating) I made can. Check it out.

**Comment by okraits** [comment]
I sometimes make a terminal window floating, but I don't want all terminal windows to be floating...

**Reply by artemave** [reply]
It tracks the last time you toggled floating for the window and applies it to new windows. If you toggle off the terminal window then all subsequent ones will be tiled.

But I agree, that this is not a good fit for windows that can be both tiled and floating. This is however a good fit for windows you almost never want tiled. E.g., arandr, pusle audio volume controle, etc.

**Reply by okraits** [reply]
```
for_window [class="XCalc"] floating enable
for_window [class="Pavucontrol"] floating enable
for_window [class="Gcolor3"] floating enable
```

For these apps I explicitely enable floating. Simple, built in, needs no installation, doesn't require additional resources :grinning:
